### DISCUSSION TITLE
Show filtered items on selectFilter

### DISCUSSION THREAD

**Discussion by GermanZun** [rappasoft/laravel-livewire-tables#941] [Q&A]
Hi!. I want to show only the items left after the table is filtered on a selectFilter. For example, i want to show only "Vanceton", "East Imaniport" and "Port Fredrick" in the 'Group City Filter" and not all of them, because it was filtered

![image](https://user-images.githubusercontent.com/12190841/195166929-cae774f8-315a-4b59-9a5a-65d8f164ff5e.png)

How can i achive this? 
Thanks!

**Comment by tonypartridge** [comment]
you would need to adjust the Builder method in the filter() function getting existing filters that are set to slim down the values. quite a trick concept if you are spanning multiple tables. 

It would make sense for the filters to have an option to limit the found date to the result filters.

**Comment by GermanZun** [comment]
I dont know how to adjust the builder in the filter function. In the filter function of the filter?

I tried to access builder on filters() method or calling getRows() so i can populate the array for the options method of filter, but it says:
 "Typed property Rappasoft\LaravelLivewireTables\DataTableComponent::$builder must not be accessed before initialization". As it says, it is not initializated at that point. How can i alter the options array after filtering?

**Reply by GermanZun** [reply]
Hi, any help about this?, is the only missing feature for me @rappasoft , thanks for you hard work.

**Comment by lrljoe** [comment]
Set a variable with the resultant IDs from the first filter, constrain the secondary filter with the variable, then refresh the filters.

Let me know if this doesn't make sense

**Reply by puntogestioncl** [reply]
Hello,
How can I update the filters???
Thanks greetings.

**Reply by lrljoe** [reply]
You'll need to give a bit more detail than that for me to be able to advise!

**Reply by puntogestioncl** [reply]
Hi lrljoe,

I have two filters, for example, one with countries and the other with states. When starting, both filters have all the data, what I need is that when selecting a country, the second filter updates and contains only the states of that country. I hope it is understood, since I am using google translate.
Greetings

**Reply by lrljoe** [reply]
So for version 3 (v3) this is pretty straight forward, for version 2 (v2) I don't have a working example just yet.

Let me know which version of the package you're on please

**Reply by puntogestioncl** [reply]
Hi,

In version "rappasoft/laravel-livewire-tables": "^2.8".

**Reply by lrljoe** [reply]
Can you get the exact installed version please, either via composer show or your composer lock file

**Reply by puntogestioncl** [reply]
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/59832219/940a4441-39b0-4822-9758-d01287cf1be9)

**Reply by lrljoe** [reply]
Definitely worth looking at updating, as there's a fair few bug fixes between 2.8 and 2.15 anyway. Do test first, but I'm not aware of any breaking changes.

I'll take a look next week at a way to backport the v3 beta example to v2.x

**Reply by puntogestioncl** [reply]
Hello lrljoe,
Where I find the V3 documentation to review the available filters, and do what I need. Thanks for your help.
Greetings.

**Reply by puntogestioncl** [reply]
Hello,

Try the following, using Livewire's hooks, and locally, I use Sail. It runs smoothly.

```php
// property
public $branches;

// in the constructor
    public function __construct()
    {
        $this->branches = $this->branchOffices(null);
    }

// in the filter 
SelectFilter::make('Sucursal')
                ->options(
                    $this->branches
                )
                ->filter(function(Builder $builder, string $value) {
                    $this->selectBranches($builder, $value);
                }),

// in the hook
public function updatedTableFiltersBrand( $value ): void
    {
        $this->branches = $this->branchOffices($value);
        $this->filters();
    }
``` 

But in production, when selecting on the first filter, the second does not change. When making a second selection in the first filter, the second one changes, but with the previously selected option. As I indicated, it only happens in production.

Greetings.

**Reply by lrljoe** [reply]
V3 Docs are presently available in GitHub by using the v3-develop branch.

I've got a few docs to write before the official site will be updated.

I'm adding a couple of features imminently and some additional docs for this specific use case to help smooth things out!

Expect it by the end of the week, but feel free to poke me on Discord if you want further info

**Reply by puntogestioncl** [reply]
Hello IrIjoe,
Thank you so much.
Greetings

**Reply by lrljoe** [reply]
No problem, do remember to poke on Discord, as I'll be able to share preview docs before I commit them for review if needed
